cmd/osquery-perf/agent.go

```diff
@@ -1034,12 +1035,17 @@ func (a *agent) config() error {
 			}
 			q.ScheduleInterval = m["interval"].(float64)
 			q.Query = m["query"].(string)
-			scheduledQueryData = append(scheduledQueryData, q)
+
+			scheduledQueryName := packName + "_" + queryName
+			if existingEntry, ok := a.scheduledQueryData[scheduledQueryName]; ok {
+				// Keep lastRun if the query is already scheduled.
+				q.lastRun = existingEntry.lastRun
+			}
+			scheduledQueryData[scheduledQueryName] = q
```

Comment:
The bug is that we were overriding nextRun with 0 at every `/api/v1/osquery/config` request (every 1 minute).

Reply:
So I'm trying to understand if that could've impacted my load test, how is this impacting CPU and memory of the Fleet instances? Would that have a significant impact on the mysql DB and if so, more on the reader or writer?

Reply:
By default the load test environment has ~500 queries, but only 4 queries with a interval != 0 (3 daily, 1 hourly). This bug was causing the 4 queries to be executed every minute (config interval is 1m) instead of their configured interval.

Am not 100% sure if this was the solely reason we experienced 100% CPU utilization with 80k hosts, we did see many log requests (that did not match the configured frequencies).